wsmaster/che-core-api-factory-bitbucket-server/src/main/java/org/eclipse/che/api/factory/server/bitbucket/BitbucketServerPersonalAccessTokenFetcher.java

```diff
@@ -103,6 +106,20 @@ public PersonalAccessToken fetchPersonalAccessToken(Subject cheUser, String scmS
   @Override
   public Optional<Boolean> isValid(PersonalAccessToken personalAccessToken)
       throws ScmCommunicationException, ScmUnauthorizedException {
+    // If BitBucket oAuth is not configured try to find a manually added user namespace token.
+    if (bitbucketServerApiClient instanceof NoopBitbucketServerApiClient
```

Comment:
I don't clear understand why `bitbucketServerApiClient instanceof NoopBitbucketServerApiClient` is even possible here

Reply:
Changed the check.